cpp/test/sg/fil_test.cu

```diff
@@ -293,10 +293,10 @@ class BaseFilTest : public testing::TestWithParam<FilTestParams> {
     // uniformily distributed in orders of magnitude: smaller models which
     // still stress large bitfields.
     // up to 10**ps.max_magnitude_of_matching_cat (only if feature is categorical, else -1)
-    cat_sets_h = cat_sets_owner(ps.num_cols, ps.num_trees);
     std::mt19937 gen(ps.seed);
     std::uniform_real_distribution mmc(-1.0f, ps.max_magnitude_of_matching_cat);
     std::bernoulli_distribution fc(ps.feature_categorical_prob);
+    cat_sets_h.max_matching.resize(ps.num_cols);
```

Comment:
I will insist on coupling the sizes with the loops that fill the elements. And since we don't port test code into the main classes (and the test code just generates things in a different way), the only other thing I imagine is generating all the arrays separately from cat_sets_owner, and exporting a treelite model before we even construct a cat_sets_owner.